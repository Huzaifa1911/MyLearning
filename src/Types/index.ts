export type TAnime = {
  mal_id: number;
  image: string | null;
  title: string;
  rating: string;
};

export type TShowLoading = {showLoading?: boolean};
