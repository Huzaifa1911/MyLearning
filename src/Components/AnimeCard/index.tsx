import {View, StyleSheet, Image, TouchableOpacity} from 'react-native';
import React from 'react';
import {TAnime} from '../../Types';
import AppText from '../Common/AppText';
import Spacer from '../Common/Spacer';
import {AppTheme} from '../../Assets/Theme';

interface IAnimeCardProps {
  anime: TAnime;
  onPress: () => void;
}

const AnimeCard = ({anime, onPress}: IAnimeCardProps) => {
  const {title, image, rating} = anime;
  const disabled = !onPress;
  return (
    <TouchableOpacity
      style={styles.card}
      activeOpacity={0.6}
      onPress={onPress}
      testID="animeCard"
      disabled={disabled}>
      <View style={styles.content}>
        <Image
          style={styles.image}
          source={{uri: image ?? ''}}
          resizeMode="cover"
        />

        <View style={styles.textContent}>
          <AppText size={18} fontWeight="600" numberOfLines={2}>
            {title}
          </AppText>
          <Spacer top={10} />
          <AppText size={14} fontWeight="400">
            {rating}
          </AppText>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default AnimeCard;

const styles = StyleSheet.create({
  card: {
    height: 100,
    paddingTop: 10,
    paddingHorizontal: 8,
    borderWidth: 0.3,
    borderRadius: 11,
    shadowColor: AppTheme.colors.backdrop,
    backgroundColor: AppTheme.colors.white,
    // shadowOpacity: 0.3,
    // shadowRadius: 5,
    // shadowOffset: {width: 0, height: 5},
  },
  textContent: {width: '70%', marginLeft: 10},
  content: {
    flexDirection: 'row',
    alignItems: 'flex-start',
  },
  image: {height: 80, width: 80, borderRadius: 10},
});
