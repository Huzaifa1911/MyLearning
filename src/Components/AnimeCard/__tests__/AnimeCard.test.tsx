/* eslint-disable react/react-in-jsx-scope */
import {render} from '@testing-library/react-native';
import AnimeCard from '..';
import {TAnime} from '../../../Types';

describe('Anime Card', () => {
  it('Renders Correctly', () => {
    // Mocking data
    const anime: TAnime = {title: 'Anime', image: null, mal_id: 2, rating: ''};
    const tree = render(<AnimeCard anime={anime} onPress={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
});
