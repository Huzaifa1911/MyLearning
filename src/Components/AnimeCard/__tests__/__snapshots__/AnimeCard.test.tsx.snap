// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anime Card Renders Correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 11,
      "borderWidth": 0.3,
      "height": 100,
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingTop": 10,
      "shadowColor": "rgba(0, 0, 0, 0.5)",
    }
  }
  testID="animeCard"
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "uri": "",
        }
      }
      style={
        {
          "borderRadius": 10,
          "height": 80,
          "width": 80,
        }
      }
    />
    <View
      style={
        {
          "marginLeft": 10,
          "width": "70%",
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "flex": undefined,
              "marginBottom": undefined,
              "marginHorizontal": undefined,
              "marginLeft": undefined,
              "marginRight": 5,
              "marginTop": undefined,
              "marginVertical": undefined,
            }
          }
        />
        <Text
          numberOfLines={2}
          style={
            {
              "color": "#18234C",
              "fontSize": 18,
              "fontWeight": "600",
              "textAlign": "left",
              "textDecorationLine": "none",
              "textTransform": "none",
            }
          }
        >
          Anime
        </Text>
        <View
          style={
            {
              "backgroundColor": "transparent",
              "flex": undefined,
              "marginBottom": undefined,
              "marginHorizontal": undefined,
              "marginLeft": 5,
              "marginRight": undefined,
              "marginTop": undefined,
              "marginVertical": undefined,
            }
          }
        />
      </View>
      <View
        style={
          {
            "backgroundColor": "transparent",
            "flex": undefined,
            "marginBottom": undefined,
            "marginHorizontal": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": 10,
            "marginVertical": undefined,
          }
        }
      />
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            {
              "backgroundColor": "transparent",
              "flex": undefined,
              "marginBottom": undefined,
              "marginHorizontal": undefined,
              "marginLeft": undefined,
              "marginRight": 5,
              "marginTop": undefined,
              "marginVertical": undefined,
            }
          }
        />
        <Text
          style={
            {
              "color": "#18234C",
              "fontSize": 14,
              "fontWeight": "400",
              "textAlign": "left",
              "textDecorationLine": "none",
              "textTransform": "none",
            }
          }
        />
        <View
          style={
            {
              "backgroundColor": "transparent",
              "flex": undefined,
              "marginBottom": undefined,
              "marginHorizontal": undefined,
              "marginLeft": 5,
              "marginRight": undefined,
              "marginTop": undefined,
              "marginVertical": undefined,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
